app/test/block_production_test.go

```diff
@@ -55,3 +57,35 @@ func (s *BlockProductionTestSuite) Test_BlockOneTransactionNonInclusion() {
 	require.Error(err) // change this to require.NoError(err) to see the error
 	require.ErrorContains(err, "rpc error: code = Unknown desc = codespace sdk code 18: invalid request: failed to load state at height 0; no commit info found (latest height: 0)")
 }
+
+// Test_FirstBlockIsEmpty tests whether the first block is empty.
+func (s *BlockProductionTestSuite) Test_FirstBlockIsEmpty() {
+	require := s.Require()
+	// wait until height 1 before posting transactions
+	// otherwise tx submission will fail
+	time.Sleep(1 * s.timeoutCommit)
+	// send some transactions, these should be included in the second block
+	_, err := s.cctx.PostData(s.accounts[0], flags.BroadcastBlock, appns.RandomBlobNamespace(), tmrand.Bytes(100000))
+	require.NoError(err)
+
+	// wait for 2*s.timeoutCommit+1*time.Second to ensure that the node is
+	// at height 2
+	_, err = s.cctx.WaitForHeightWithTimeout(2, 2*s.timeoutCommit+1*time.Second)
```

Comment:
[question] if this only cares about waiting for block height 2, is that specific time necessary? Very minor but It's difficult to read. Put another way, this seems to work:

```suggestion
	// Wait for block height 2.
	_, err = s.cctx.WaitForHeightWithTimeout(2, s.timeoutCommit)
	require.NoError(err)
```

Reply:
Thanks, your comment is addressed in this [PR](https://github.com/celestiaorg/celestia-app/pull/2284).